### PR TITLE
DOC-9567 - Index Management with Scopes and Collections Support

### PR DESCRIPTION
**Pull request by maria-robobug**
- Adds missing Index Management examples and support for scopes and collections
- Adds note for missing Management implementations in SDK 4.0
- Adds missing User Management page and examples

### CODE REVIEW

File: modules/howtos/pages/provisioning-cluster-resources.adoc
Comment (maria-robobug) [inline modules/howtos/pages/provisioning-cluster-resources.adoc:53]: I added a note here so users are aware that these managers are not fully implemented yet.
@avsej I mentioned this would be fixed in a future `4.x` release, does this seem correct?

File: modules/howtos/pages/provisioning-cluster-resources.adoc
Comment (RichardSmedley) [inline modules/howtos/pages/provisioning-cluster-resources.adoc:8]: Are these something close to overoptimising for d.r.y.?  ;)

**Review by RichardSmedley — APPROVED** [review]
Great to see these up - I'm going to leave my plus one here for the docs, but let's wait for @avsej 's code review before merging.  :+1:

File: modules/howtos/pages/provisioning-cluster-resources.adoc
Comment (maria-robobug) [reply modules/howtos/pages/provisioning-cluster-resources.adoc:8]: We use these in sdk-common as well, so wanted to avoid repeating myself there 😄 .

**Review by avsej — APPROVED** [review]

**Pull request merged**
